shared/src/main/scala/bloop/io/Environment.scala

```diff
@@ -0,0 +1,174 @@
+package bloop.io
+
+object Environment {
+
+  /**
+   * Returns "\n" if both OSTYPE and SHELL are nonEmpty, returns `System.lineSeparator` otherwise.
+   * @see [[shellLineSeparator]] which is similar.
+   */
+  def lineSeparator: String = if (validOstype.nonEmpty) shellLineSeparator else System.lineSeparator
```

Comment:
This way we are essentially checking both `OSTYPE` and `SHELL` - is both needed to be set?

Reply:
The difference is that `EOL` and `lineSeparator` might have different values, depending on whether both `OSTYPE` and `SHELL` are validated. 

My intent was that `EOL` should be `'\n'` in a valid `SHELL` environment, whether `OSTYPE` is valid or not.
That fixes the autocompletion bug,  since `EOL` is used as the line ending of `BloopLogger#printInfo`.

If `OSTYPE` is undefined (or not recognized as a valid target OS), then `lineSeparator` will equal `System.lineSeparator`, so most output line endings, other than `BloopLogger#printInfo` will, be unchanged.

Reply:
Maybe just:
```suggestion
  def lineSeparator: String = if (validOstype.nonEmpty && validShell.nonEmpty) shellLineSeparator else sysLineSeparator
```
shellLineSeparator is only used for EOL, which itself is only used in one place.